creator/projects/schema.py

```diff
@@ -0,0 +1,164 @@
+import requests
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AnonymousUser
+from graphql import GraphQLError
+from graphene import (
+    relay,
+    Boolean,
+    InputObjectType,
+    ObjectType,
+    Field,
+    String,
+    Mutation,
+)
+from graphene_django import DjangoObjectType
+from graphene_django.filter import DjangoFilterConnectionField
+
+from .models import CavaticaProject
+
+
+class CavaticaProjectNode(DjangoObjectType):
+    class Meta:
+        model = CavaticaProject
+        filter_fields = ["name"]
+        interfaces = (relay.Node,)
+
+    @classmethod
+    def get_node(cls, info, project_id):
+        """
+        Only return node if user is Cavatica data scientist
+        """
+        try:
+            cavaticaProject = cls._meta.model.objects.get(project_id=project_id)
+        except cls._meta.model.DoesNotExist:
+            return None
+
+        user = info.context.user
+
+        if not user.is_authenticated:
+            return None
+
+        if user.is_admin:
+            return cavaticaProject
+
+        return None
+
+
+class CavaticaProjectMutation(graphene.Mutation):
+    """
+    Updates fields for a given Cavatica project.
+    """
+
+    class Arguments:
+        project_id = graphene.String(required=True)
+        name = graphene.String()
+        description = graphene.String()
+
+    cavaticaProject = graphene.Field(CavaticaProjectNode)
+
+    def mutate(self, info, kf_id, **kwargs):
+        """
+        Updates an existing Cavatica project.
+        User must be authenticated and be ADMIN.
+        """
+        user = info.context.user
+        if user is None or not user.is_authenticated:
+            raise GraphQLError("Not authenticated to mutate a Cavatica project.")
+
+        try:
+            cavaticaProject = CavaticaProject.objects.get(project_id=project_id)
+        except CavaticaProject.DoesNotExist:
+            raise GraphQLError("Cavatica project does not exist.")
+
+        if "ADMIN" not in user.ego_roles:
```

Comment:
Probably better to do the auth check as soon as possible and before looking up the object in the database.